### DISCUSSION TITLE
Does quarkus support ContainerRequestFilter in a project dependency?

### DISCUSSION THREAD

**Discussion by newk5** [quarkusio/quarkus#24133] [Q&A] [answered]
I have a maven project running quarkus with resteasy, and that project has another maven dependency (using  quarkus dependencies too), and that project has 1 class which implements ContainerRequestFilter but the filter method is never triggered no matter how many requests are made. In addition to that, it completely breaks Hot-reload in dev-mode for the the whole app, meaning, no matter how many changes I make to my code, it nevers triggers a reload.

The structure looks like this:

- Main project (contains JAX-RS endpoints, this is where I run: quarkus:dev)
  - Main project internals (contains the ContainerRequestFilter )
  
If I move the ContainerRequestFilter to the main project, everything starts working again. 

Is this not supported in quarkus or is it a bug?

**Comment by glefloch** [comment, marked as answer]
By default dependencies are not scanned for beans. 

You can add an empty `META-INF/beans.xml` file in your dependency resource directory to make sure it will be scanned. Here is the docs explaining it: https://quarkus.io/guides/cdi-reference#bean_discovery

**Reply by newk5** [reply]
Thank you, that was the issue, all working now 👍
